ssl/ssl_ciph.c

```diff
@@ -465,7 +465,8 @@ DEFINE_RUN_ONCE_STATIC(do_load_builtin_compressions)
             comp->method = method;
             comp->id = SSL_COMP_ZLIB_IDX;
             comp->name = COMP_get_name(method);
-            sk_SSL_COMP_push(ssl_comp_methods, comp);
+            if (!sk_SSL_COMP_push(ssl_comp_methods, comp))
+                OPENSSL_free(comp);
             sk_SSL_COMP_sort(ssl_comp_methods);
```

Comment:
Is the sort still needed if the push fails?

Reply:
no, but it does not hurt either.  So I did not want to complicate things further,
since this PR does not claim to fix the error handling here.
A proper error handling should return 0 if the function did not succeed
and probably clean up things when they were not constructed properly.
But that style of missing error handling can be found by my error injection
method pretty much everywhere, therefore I decided to concentrate on
memory leaks and possible crashes...